src/tof/chopper.py

```diff
@@ -76,19 +81,43 @@ def open_close_times(
         """
         if unit is None:
             unit = time_limit.unit
-        nrot = max(int(sc.ceil((time_limit * self.frequency).to(unit='')).value), 1)
+        nrot = max(
+            int(sc.ceil((time_limit * abs(self.frequency)).to(unit='')).value), 1
+        )
         # Start at -1 to catch early openings in case the phase or opening angles are
         # large
         phases = sc.arange(uuid.uuid4().hex, -1, nrot) * two_pi + self.phase.to(
             unit='rad'
-        )
+        ) * (np.sign(self.frequency.value) * -1.0)
         # Note that the order is important here: we need (phases + open/close) to get
         # the correct dimension order when we flatten below.
-        open_times = (phases + self.open.to(unit='rad', copy=False)) / self.omega
-        close_times = (phases + self.close.to(unit='rad', copy=False)) / self.omega
+        open_times = (phases + self.open.to(unit='rad', copy=False)).flatten(
+            to=self.open.dim
+        )
+        close_times = (phases + self.close.to(unit='rad', copy=False)).flatten(
+            to=self.close.dim
+        )
+        # If the chopper is rotating anti-clockwise, we mirror the openings because the
+        # first cutout will be the last to open.
+        if self.frequency.value > 0:
+            open_times, close_times = (
+                sc.array(
+                    dims=close_times.dims,
+                    values=(two_pi - close_times).values[::-1],
+                    unit=close_times.unit,
+                ),
+                sc.array(
+                    dims=open_times.dims,
+                    values=(two_pi - open_times).values[::-1],
+                    unit=open_times.unit,
+                ),
+            )
```

Comment:
The interface and need to handle this here feels very error-prone. Would it be possible to always specify openings in the order they appear, with positive angles? Any code for converting from different convention could be moved into helpers, such as `Angles.from_nexus(**nexus_chopper_angles)`.

Reply:
Nice suggestion, but wouldn't this mean that the `direction` parameter would then be obsolete? Because we would always have the first opening crossing the beam first, regardless of the direction of rotation?

So I am wondering if it could me more error-prone in some cases?

Reply:
Not sure how different conventions for different things (rotation, phase, angles) interact, so this was just an idea. I don't mind either solution, but I thought that `direction` alone might not be sufficient, because there may be other conventions as well?

Reply:
Hmm, now that I think a bit more about this, I am wondering if we should do what you suggest, and not have a sign on the frequency, nor a `direction` parameter...

In the case of the dream choppers, there are two counter-rotating choppers, where one is above the beam and the other one is below. There is even a third chopper which is on the side:
![Screenshot at 2023-08-07 17-51-31](https://github.com/scipp/tof/assets/39047984/cbca066a-996f-4d60-8e8e-ffc255f01662)

- NeXus defines a TDC on the chopper, and an offset between the TDC and the position of the beam.
- McStas deals with this by defining the centre of the beam as the zero angle, meaning that the chopper above the beam is rotated 180deg compared to the one below.

I think defining things in terms of "the first window that is listed is the first one that will pass in front of the beam" makes sense. If a user has angles from a chopper diagram, and then they want to place that chopper on the side of the beam, they just have to either add 90deg to all their angles, or add 90deg to their phase.

Reply:
> Does this mean we have to change nothing to the code but instead update the documentation to explain how to make a counter-rotating chopper?

I think we probably still need a `direction` parameter. Users typically have a diagram giving the opening angles:
![Screenshot at 2023-08-07 18-46-27](https://github.com/scipp/tof/assets/39047984/99cf68ee-2d1b-4c3c-9ea6-72dff92964f5)

It will not always be `from_nexus` or `from_mcstas`, sometimes you just want to check the values given in a paper. I think it will be useful to be able to set `direction='anticlockwise'`.

Reply:
About `Angles.from_nexus(**nexus_chopper_angles)`, wouldn't we try and load everything from NeXus, not just the angles but the frequency, the phase etc?

For McStas, we cannot parse the instrument file, so I am not sure what to do there.

I am asking because I thought about either having `Angles.counter_rotating(**angles)`, or a method on the chopper to flip the rotation, e.g.
```Py
chopper = Chopper(...)
chopper.counter_rotate()  # would modify in-place the open and close angles
```

The problem with the latter is that you don't necessarily remember how many times you called `counter_rotate()` and if you are doing stuff in the notebook, you may accidentally flip you chopper again, so it would probably be better to have is set when constructing the chopper.

How about using classmethods?
- `Chopper.anti_clockwise(...)`
- `Chopper.from_nexus(...)`

Reply:
👍 on the classmethods, this is more or less what I had in mind.